egs/librispeech/ASR/conformer_ctc/train.py

```diff
@@ -606,6 +606,35 @@ def run(rank, world_size, args):
     train_dl = librispeech.train_dataloaders()
     valid_dl = librispeech.valid_dataloaders()
 
+    from lhotse.dataset import find_pessimistic_batches
+
+    logging.info(
+        "Sanity check -- see if any of the batches in epoch 0 would cause OOM."
+    )
+    batches, crit_values = find_pessimistic_batches(train_dl.sampler)
+    for criterion, cuts in batches.items():
+        logging.info(
+            f"* criterion: {criterion} (={crit_values[criterion]}) ..."
+        )
+        batch = train_dl.dataset[cuts]
+        try:
+            compute_loss(
```

Comment:
I can update the model for the batches, I doubt it would break anything, and as you mentioned this will be more accurate. This should be sufficient, right?

```
loss.backward()
clip_grad_norm_(model.parameters(), 5.0, 2.0)
optimizer.step()
optimizer.zero_grad()
```

Reply:
Yes, this would be more accurate.

But I doubt whether `find_pessimistic_batches` will work. A sampler sees different batches in different epochs. And when multi-GPU training comes into scope, some batches are skipped for some sampler in epoch 0, but will be seen in other epochs.

I am not sure whether the value of max duration obtained from epoch 0 can be applied to other epochs.